src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java

```diff
@@ -939,6 +939,56 @@ public void testOutOfRangesArguments(final String[] argv) throws Exception {
         return result;
     }
 
+    @CommandLineProgramProperties(
+            summary = "tool with bounded collection argument",
+            oneLineSummary = "bounded collection argumen",
+            programGroup = TestProgramGroup.class
+    )
+    public class BoundedCollection {
+        @Argument(doc = "Integer collection in the range [0, 30]",
+                optional = true,
+                minValue = 0, maxValue = 30,
+                minRecommendedValue = 10, maxRecommendedValue = 15)
+        public List<Integer> intListArg = new ArrayList<>();
```

Comment:
And what will happen if there are default values and `--intListArg null --intListArg 15` to clean the defaults?